### DISCUSSION TITLE
Istio mTLS from Ingress to Sidecar

### DISCUSSION THREAD

**Discussion by abasitt** [istio/istio#48270] [Q&A]
Hi All,

I was troubleshooting an external_auth service in Istio ingress which was implemented using envoyfilter and the keycloak destination service is also locally deployed in the cluster.
What I noticed is that no connection request was coming to the keycloak and later I found out it was because of mTLS strict mode.

Is this correct understanding ? I think the answer is [here ](https://istio.io/latest/docs/ops/configuration/traffic-management/tls-configuration/) but would like to double confirm.
If the request is originating from the Ingress gateway, it will not be sent as mTLS and any external auth destination will need a DR to explicitly configure mTLS ?
If the request terminated at the ingress gateway with final destination is a service within a mesh that will be auto mTLS ?

**Comment by linsun** [comment]
Depends on if your keycloak is in the mesh?  If it is in the mesh, Istio usually auto mTLS.

**Reply by irisdingbj** [reply]
yep, if the service is in mesh - (like it has a sidecar injected),  by default the communication between the gateway and the in mesh service will be mTLS.

**Reply by abasitt** [reply]
thanks Lin and Iris. This is what I initially thought as well but it isn't the behavior what I observed.
The auto-mtls is working fine for everything from ingress-gateway except if the traffic is initiated to destinations in the envoyfilter. 
Keycloak is part of the mesh with sidecar.

**Reply by abasitt** [reply]
Istio version 1.17.3

**Comment by irisdingbj** [comment]
yep, if the service is in mesh - (like it has a sidecar injected), by default the communication between the gateway and the in mesh service will be mTLS.
